docker-compose.yml

```diff
@@ -109,27 +119,42 @@ services:
     volumes:
       - vault-file:/vault/file
       - vault-logs:/vault/logs
+      - vault-init:/vault/init:ro
+      - /tmp/edgex/secrets/edgex-vault:/tmp/edgex/secrets/edgex-vault:ro
+    depends_on:
+      - security-secrets-setup
+      - consul
+
+  security-secrets-setup:
+    image: ${securitySecretsSetup}
+    container_name: edgex-secrets-setup
+    hostname: edgex-secrets-setup
+    tmpfs:
+      - /tmp
+    command: "generate"
+    volumes:
       - secrets-setup-cache:/etc/edgex/pki
-      - vault-config:/vault/config
+      - vault-init:/vault/init
+      - /tmp/edgex/secrets:/tmp/edgex/secrets
     depends_on:
       - volume
-      - consul
 
   vault-worker:
     image: ${vaultWorker}
     container_name: edgex-vault-worker
     hostname: edgex-vault-worker
-    command: ["--init=true", "--vaultInterval=10", "--insecureSkipVerify=true"]
     networks:
       edgex-network:
         aliases:
           - edgex-vault-worker
     volumes:
       - vault-config:/vault/config
+      - consul-scripts:/consul/scripts
+      - /tmp/edgex/secrets/ca/ca.pem:/tmp/edgex/secrets/ca/ca.pem:ro
+      - /tmp/edgex/secrets/edgex-kong:/tmp/edgex/secrets/edgex-kong:ro
     depends_on:
       - volume
       - consul
-      - vault
```

Comment:
This dependency should stay since we need `Vault` to start up first

Reply:
Back in.